execute/tokendata/usdc/http.go

```diff
@@ -98,83 +98,89 @@ func (r httpResponse) validate() error {
-			return empty, http.StatusTooManyRequests, ErrRateLimit
+			return nil, http.StatusTooManyRequests, ErrRateLimit
 		}
 	}
 
+	// Use a timeout to guard against attestation API hanging, causing observation timeout and
+	// failing to make any progress.
 	timeoutCtx, cancel := context.WithTimeoutCause(ctx, h.apiTimeout, ErrTimeout)
 	defer cancel()
 
 	requestURL := *h.apiURL
-	requestURL.Path = path.Join(requestURL.Path, apiVersion, attestationPath, fmt.Sprintf("0x%x", messageHash))
+	requestURL.Path = path.Join(requestURL.Path, apiVersion, attestationPath, messageHash.String())
 	return h.callAPI(timeoutCtx, requestURL)
 }
 
-func (h *httpClient) callAPI(ctx context.Context, url url.URL) ([]byte, int, error) {
-	var response []byte
-	// Use a timeout to guard against attestation API hanging, causing observation timeout and
-	// failing to make any progress.
+func (h *httpClient) callAPI(ctx context.Context, url url.URL) (cciptypes.Bytes, HTTPStatus, error) {
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url.String(), nil)
 	if err != nil {
-		return response, http.StatusBadRequest, err
+		return nil, http.StatusBadRequest, err
 	}
 	req.Header.Add("accept", "application/json")
 	res, err := http.DefaultClient.Do(req)
 	if err != nil {
 		if errors.Is(err, context.DeadlineExceeded) {
-			return response, http.StatusRequestTimeout, ErrTimeout
+			return nil, http.StatusRequestTimeout, ErrTimeout
 		}
 		// On error, res is nil in most cases, do not read res.StatusCode, return BadRequest
-		return response, http.StatusBadRequest, err
+		return nil, http.StatusBadRequest, err
 	}
-	defer res.Body.Close()
 
+	var status HTTPStatus
+	if res != nil {
+		defer res.Body.Close()
+		status = HTTPStatus(res.StatusCode)
+	}
 	// Explicitly signal if the API is being rate limited
 	if res.StatusCode == http.StatusTooManyRequests {
 		h.setCoolDownPeriod(res.Header)
-		return response, res.StatusCode, ErrRateLimit
+		return nil, status, ErrRateLimit
 	}
 	if res.StatusCode == http.StatusNotFound {
-		return response, res.StatusCode, ErrNotReady
+		return nil, status, ErrNotReady
 	}
 	if res.StatusCode != http.StatusOK {
-		return response, res.StatusCode, ErrUnknownResponse
+		return nil, status, ErrUnknownResponse
 	}
 
-	return h.parsePayload(res)
+	response, err := h.parsePayload(res)
+	return response, status, err
 }
 
-func (h *httpClient) parsePayload(res *http.Response) ([]byte, int, error) {
+func (h *httpClient) parsePayload(res *http.Response) (cciptypes.Bytes, error) {
+	if res == nil {
+		return nil, ErrUnknownResponse
+	}
+
 	var response httpResponse
 	if err := json.NewDecoder(res.Body).Decode(&response); err != nil {
-		return nil, res.StatusCode, fmt.Errorf("failed to decode json: %w", err)
+		return nil, fmt.Errorf("failed to decode json: %w", err)
 	}
 
-	if err1 := response.validate(); err1 != nil {
-		return nil, res.StatusCode, err1
+	if err := response.validate(); err != nil {
+		return nil, err
```

Comment:
wrap err

Reply:
validate has already nicely formatter error, no need to wrap